pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/NodeInfoPanelController.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner;
+
+import java.net.URL;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.ResourceBundle;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.ast.xpath.Attribute;
+import net.sourceforge.pmd.lang.ast.xpath.AttributeAxisIterator;
+import net.sourceforge.pmd.lang.java.ast.TypeNode;
+import net.sourceforge.pmd.lang.symboltable.NameDeclaration;
+import net.sourceforge.pmd.util.fxdesigner.model.MetricEvaluator;
+import net.sourceforge.pmd.util.fxdesigner.model.MetricResult;
+import net.sourceforge.pmd.util.fxdesigner.util.controls.MetricResultListCell;
+import net.sourceforge.pmd.util.fxdesigner.util.controls.ScopeHierarchyTreeCell;
+import net.sourceforge.pmd.util.fxdesigner.util.controls.ScopeHierarchyTreeItem;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.control.Tab;
+import javafx.scene.control.TabPane;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeView;
+
+
+/**
+ * Controller of the node info panel (left).
+ *
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public class NodeInfoPanelController implements Initializable {
+
+    private final DesignerRoot designerRoot;
+    private final MainDesignerController parent;
+
+    @FXML
+    private TabPane nodeInfoTabPane;
+    @FXML
+    private Tab xpathAttributesTitledPane;
+    @FXML
+    private ListView<String> xpathAttributesListView;
+    @FXML
+    private Tab metricResultsTitledPane;
+    @FXML
+    private ListView<MetricResult> metricResultsListView;
+    @FXML
+    private Label metricsTitleLabel;
+    @FXML
+    private TreeView<Object> scopeHierarchyTreeView;
+    private MetricEvaluator metricEvaluator = new MetricEvaluator();
+
+
+    NodeInfoPanelController(DesignerRoot root, MainDesignerController mainController) {
+        this.designerRoot = root;
+        parent = mainController;
+    }
+
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        metricResultsListView.setCellFactory(param -> new MetricResultListCell());
```

Comment:
In theory, these cell factories can be set directly in fxml:

```
<?import net.sourceforge.pmd.util.fxdesigner.util.controls.MetricResultListCell?>
<?import net.sourceforge.pmd.util.fxdesigner.util.controls.ScopeHierarchyTreeCell?>

<ListView fx:id="metricResultsListView" styleClass="secondary-panel" BorderPane.alignment="CENTER">
    <cellFactory>
        <MetricResultListCell />
    </cellFactory>
</ListView>

<TreeView fx:id="scopeHierarchyTreeView" styleClass="secondary-panel" BorderPane.alignment="CENTER">
    <cellFactory>
        <ScopeHierarchyTreeCell />
    </cellFactory>
</TreeView>
```

Reply:
That doesn't work... I've also tried doing using the [fx:factory](https://docs.oracle.com/javafx/2/api/javafx/fxml/doc-files/introduction_to_fxml.html) attribute, and a custom callback factory, but no way... it's a shame

Reply:
Entschuldigung, it actually works. IntelliJ just didn't refresh the resource file cache, ignoring changes to the fxml files ^^